main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -441,6 +441,19 @@ object Redundancy {
       val us3 = visitExp(exp3, env0, rc)
       us1 ++ us2 ++ us3
 
+    case Expr.While(exp1, exp2, _, _, _) =>
+      val us1 = visitExp(exp1, env0, rc)
+      val us2 = visitExp(exp2, env0, rc)
+      val pureCondition = exp1 match {
```

Comment:
This is the only interesting part. Should we warn users about pure conditions? `while (true) ..` might be used when e.g. an effect will break the loop or the loop is truly infinite (server perhaps).

This version allows `while(true)` but doesn't allow `while(0 < 1)` since that is probably user error